### ISSUE TITLE
include pandas, matplotlib and other bundled stubs 

### ISSUE THREAD

**Issue by tekumara** [microsoft/pyright#861]
Pylance bundles stubs for:
* django
* jmespath
* matplotlib
* pandas
* PIL
* onnxruntime

Pyright doesn't.

Would be nice to have a consistent user experience when using Pylance in VSCode and Pyright on the CLI and in CI/CD.

If this isn't possible, would it be possible to provide these stubs (in particular pandas and matplotlib) as a separately installable pypi distribution?

**Comment by erictraut** [comment]
Thanks for the suggestion. There are some licensing questions/details to work out, but I agree with the goal that you've stated above, and we're working to address that goal. It will probably not involve a separately installable pypi distribution, however.

**Comment by m3at** [comment]
@tekumara in the meanwhile you can find stubs for numpy/pandas/matplotlib [here](https://github.com/predictive-analytics-lab/data-science-types) (install with `pip install data-science-types`).

**Comment by fabiencelier** [comment]
+1 for pandas and numpy
What's the recommended way to use data-science-types in my project with pyright ?
Should I clone the relevant library stubs into the typings folder of my project or use it as a Python dependency ?

**Comment by erictraut** [comment]
For libraries that include stubs, install them as you would normally. In some cases, you will need to update to the latest package version. Numpy, for example, has only recently added in-package stubs.

For libraries that publish company "stub packages" (whose names end in "-stubs"), install those into your Python environment.

For other libraries that do not yet publish their own stubs, if you can find stubs (or want to create minimal versions of stubs yourself), you can place these in your local stubs folder. Depending on your project, you may want to check these in to your repo so other members of your team have access to them as well.

**Comment by erictraut** [comment]
A quick update on this request. The stub files that ship with Pylance are very incomplete. They're sufficient for basic type completions, but they are not complete enough for strict type checking or for submission to typeshed. Additional work needs to be done on them first. Once each stub is far enough along, it will be submitted to typeshed and get incorporated into pyright.

**Comment by blumu** [comment]
@m3at Unfortunately the `data-science-types` bundle multiple packages together, and in particular it overrides the `numpy` typings now provided by the official `numpy` package.

**Comment by m3at** [comment]
@blumu yes it's not ideal, having the official packages provide their own types would be best. Though you can easily use the pandas and matplotlib stubs only: fork the repo, remove [this line](https://github.com/predictive-analytics-lab/data-science-types/blob/master/setup.py#L37) and `pip install git+https://github.com/blumu/data-science-types`
